src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java

```diff
@@ -64,12 +65,21 @@ public void onCompleted(AbstractBuild build, TaskListener listener) {
 
         String additionalComment = "";
 
-        StashPostBuildCommentAction comments = build.getAction(StashPostBuildCommentAction.class);
+        StashPostBuildComment comments =
+                (StashPostBuildComment) build.getProject().getPublishersList().get(StashPostBuildComment.class);
+
         if(comments != null) {
             String buildComment = result == Result.SUCCESS ? comments.getBuildSuccessfulComment() : comments.getBuildFailedComment();
 
             if(buildComment != null && !buildComment.isEmpty()) {
-              additionalComment = "\n\n" + buildComment;
+                String expandedComment;
+                try {
+                    expandedComment = Util.fixEmptyAndTrim(build.getEnvironment(listener).expand(buildComment));
+                } catch (Exception e) {
```

Comment:
Can we get rid of this? `expand()` doesn't declare any checked exceptions, and catching `Exception` is generally discouraged

Reply:
`AbstractBuild#getEnvironment()` throws IOException and InterruptedException, so they must be caught. I agree that catching `Exception` is bad, I copied that from `StashPostBuildComment#perform()` without giving it a thought. Fixing it now.